File: Official Windows Platform Sample/Windows 8 app samples/[C#]-Windows 8 app samples/C#/Windows 8 app samples/StorageDataSource and GetVirtualizedFilesVector sample (Windows 8)/C#/ThumbnailConverter.cs
using System;
using Windows.Storage.Streams;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media.Imaging;

namespace DataSourceAdapter
{
    internal class ThumbnailConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string culture)
        {
            if (value != null)
            {
                var thumbnailStream = (IRandomAccessStream)value;
                var image = new BitmapImage();
                image.SetSource(thumbnailStream);

                return image;
            }

            return DependencyProperty.UnsetValue;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: Official Windows Platform Sample/Windows 8 app samples/[VB]-Windows 8 app samples/VB/Windows 8 app samples/StorageDataSource and GetVirtualizedFilesVector sample (Windows 8)/VB/ThumbnailConverter.vb
Imports System
Imports Windows.Storage.Streams
Imports Windows.UI.Xaml
Imports Windows.UI.Xaml.Data
Imports Windows.UI.Xaml.Media.Imaging

Friend Class ThumbnailConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As String) As Object Implements IValueConverter.Convert
        If value IsNot Nothing Then
            Dim thumbnailStream = DirectCast(value, IRandomAccessStream)
            Dim image = New BitmapImage()
            image.SetSource(thumbnailStream)

            Return image
        End If

        Return DependencyProperty.UnsetValue
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class
